src/fft.rg

```diff
@@ -172,105 +241,279 @@ function fft.generate_fft_interface(itype, dtype)
+    -- want offset_3/offset_1.
+    var offset_1 = offset_in[0].offset
+    var offset_2 = offset_in[1].offset
+    var offset_3 = offset_in[2].offset
+
+    var i_dist : int
+    if dim == 2 then
+      i_dist = offset_2/offset_1
+    elseif dim == 3 then
+      i_dist = offset_3/offset_1
+    elseif dim == 4 then
+      i_dist = offset_in[3].offset/offset_1
+    end
+
+    var istride = offset_1 / dtype_in_size
+
     var lo = input.ispace.bounds.lo:to_point()
     var hi = input.ispace.bounds.hi:to_point()
-    var flags = fftw_c.FFTW_MEASURE
-    p.p = plan_dft(
-      [data.range(dim):map(function(i) return rexpr hi.x[i] - lo.x[i] + 1 end end)],
-      [&fftw_c.fftw_complex](input_base),
-      [&fftw_c.fftw_complex](output_base),
-      fftw_c.FFTW_FORWARD,
-      flags)
+
     p.address_space = address_space
-    ;[(function()
-         if use_cuda then
-           return rquote
-             if iface.get_num_local_gpus() > 0 then
-               p.cufft_p = make_plan_gpu(input, output, address_space)
-             end
-           end
-         else
-           return rquote end
-         end
-       end)()]
+
+    var n : int[dim]
+    ;[data.range(dim):map(function(i) return rquote n[i] = hi.x[i] - lo.x[i] + 1 end end)]
+
+    -- For batched transforms, we want to exclude the last dimension as that is
+    -- the number of batches.
+    var n_batch : int[dim-1]
+    for i = 0, dim do
+      n_batch[i] = n[i]
+    end
+
+    if float_to_complex32_transform then
+      p.float_p = fftw_c.fftwf_plan_dft_r2c(dim, &n[0], [&float](input_base), [&fftw_c.fftwf_complex](output_base), fftw_c.FFTW_ESTIMATE)
+    elseif complex32_to_complex32_transform then
+      p.float_p = fftw_c.fftwf_plan_dft(dim, &n[0], [&fftw_c.fftwf_complex](input_base), [&fftw_c.fftwf_complex](output_base), fftw_c.FFTW_FORWARD, fftw_c.FFTW_ESTIMATE)
+    elseif double_to_complex64_transform then
+      p.p = fftw_c.fftw_plan_many_dft_r2c(dim-1, &n_batch[0], n[dim-1], [&double](input_base), &n_batch[0], istride, i_dist, [&fftw_c.fftw_complex](output_base), &n_batch[0], istride, i_dist, fftw_c.FFTW_ESTIMATE)
+    elseif complex64_to_complex64_transform then
+      p.p = fftw_c.fftw_plan_many_dft(dim-1, &n_batch[0], n[dim-1], [&fftw_c.fftw_complex](input_base), &n_batch[0], istride, i_dist, [&fftw_c.fftw_complex](output_base), &n_batch[0], istride, i_dist,  fftw_c.FFTW_FORWARD, fftw_c.FFTW_ESTIMATE)
+    end
+
+    rescape
+      if gpu_available then
+        remit rquote
+          if iface.get_num_local_gpus() > 0 then
+            make_plan_gpu_batch(input, output, plan, p.address_space)
+          end
+        end
+      end
+    end
   end
 
-  task iface.make_plan_task(input : region(ispace(itype), dtype),
-                            output : region(ispace(itype), dtype),
+  --- Make plan: task version. As make_plan is a __demand(__inline) task, we provide make_plan_task as a wrapper for convenience, should the user wish to use a new task and not an inlined one.
```

Comment:
```suggestion
  --- Creates the plan. This version will launch a task.
```

What I'm doing in the suggestion is make the documentation consistent with other versions and shortening to focus on what users need to know, without extraneous details.